coral-trino/src/main/java/com/linkedin/coral/trino/trino2rel/TrinoSqlValidator.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2017-2022 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.trino2rel;
+
+import org.apache.calcite.adapter.java.JavaTypeFactory;
+import org.apache.calcite.config.NullCollation;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlInsert;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperatorTable;
+import org.apache.calcite.sql.SqlUtil;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.validate.SqlConformance;
+import org.apache.calcite.sql.validate.SqlValidatorImpl;
+import org.apache.calcite.sql.validate.SqlValidatorScope;
+
+import com.linkedin.coral.common.functions.FunctionFieldReferenceOperator;
+
+
+/**
+ * Validates the parse tree of a Trino SQL statement,
+ * and provides semantic information about the parse tree.
+ */
+class TrinoSqlValidator extends SqlValidatorImpl {
```

Comment:
If the validator now works on Coral IR, why are we copying the Hive validator?

Reply:
Using `HiveSqlValidator` in coral-trino, will delete `TrinoSqlValidator` for now since it's the same code.

Reply:
Do you mean you will delete it as part of this PR?